### PR TITLE
Grafana oauth test

### PR DESCRIPTION
**Pull request by rakesh-garimella**
**Description**

Changes proposed in this pull request:

- Added automated stests for Grafana oauth2 tests
- Make the tests work for both kyma 1 and kyma 2
**Related issue(s)**

### CODE REVIEW

File: tests/fast-integration/monitoring/helpers.js [hunk 1]
```diff
@@ -110,9 +122,135 @@ async function assertTimeSeriesExist(metric, labels) {
...
+    type: "Opaque",
+    data: {
+      OAUTH2_PROXY_SKIP_PROVIDER_BUTTON: toBase64("true")
+    },
+  }
+  console.log("Creating secret: monitoring-auth-proxy-grafana-user ")
+  await k8sApply([sec], "kyma-system");
+}
+
+async function updateProxyDeployment() {
```
Comment (chrkl) [inline tests/fast-integration/monitoring/helpers.js:179]: I am wondering if we should configure at least the trusted-ip range in the secret to get as close as possible to the way a user would configure this.

File: tests/fast-integration/monitoring/client.js [hunk 2]
```diff
@@ -49,10 +49,38 @@ async function get(path) {
     return responseBody;
 }
 
+async function queryGrafana(url, redirectURL, httpErrorCode) {
+    try {
+        delete axios.defaults.headers.common["Accept"]
```
[the remark this answers is not in the document]
Comment (rakesh-garimella) [reply tests/fast-integration/monitoring/client.js:57]: fixed.

File: tests/fast-integration/monitoring/helpers.js (hunk 1 above)
Comment (rakesh-garimella) [reply tests/fast-integration/monitoring/helpers.js]: I tried it with passing the trusted IP in secret it did not work out

File: tests/fast-integration/monitoring/client.js [hunk 3]
```diff
@@ -49,10 +51,43 @@ async function get(path) {
     return responseBody;
 }
 
+async function queryGrafana(url, redirectURL, ignoreSSl, httpErrorCode) {
```
Comment (crabtree) [inline tests/fast-integration/monitoring/client.js:54]:
```suggestion
async function queryGrafana(url, redirectURL, ignoreSSL, httpErrorCode) {
```

File: tests/fast-integration/monitoring/client.js [hunk 4]
```diff
@@ -49,10 +51,43 @@ async function get(path) {
...
+    try {
+        // For more details see here: https://oauth2-proxy.github.io/oauth2-proxy/docs/behaviour
+        delete axios.defaults.headers.common["Accept"]
+        // Ignore SSL certificate for self signed certificates
+        const agent = new https.Agent({
+            rejectUnauthorized: !ignoreSSl
```
Comment (crabtree) [inline tests/fast-integration/monitoring/client.js:60]:
```suggestion
            rejectUnauthorized: !ignoreSSL
```

File: tests/fast-integration/monitoring-test/grafana-test.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    this.timeout(30 * 60 * 1000); // 30 min
+    this.slow(5 * 1000);
+
+    if (process.env.isKyma2 === "true") {
+      it("Checks grafana redirect to kyma docs", async () => {
+        let res = await assertGrafanaredirect("https://kyma-project.io/docs")
+        assert.isTrue(res, "Grafana redirect to kyma docs does not work!");
+      })
+
+      it ("Creates secret for atuh proxy redirect", async() => {
```
[the remark this answers is not in the document]
Comment (rakesh-garimella) [reply tests/fast-integration/monitoring-test/grafana-test.js]: fixed

File: tests/fast-integration/monitoring/helpers.js [hunk 6]
```diff
@@ -110,9 +123,184 @@ async function assertTimeSeriesExist(metric, labels) {
...
+    if (res === true) {
+      return res
+    }
+    await sleep(5*1000)
+    retries++
+  }
+  return false
+}
+
+async function assertGrafanaredirect(redirectURL) {
```
Comment (crabtree) [inline tests/fast-integration/monitoring/helpers.js:139]:
```suggestion
async function assertGrafanaRedirect(redirectURL) {
```

File: tests/fast-integration/monitoring/helpers.js [hunk 7]
```diff
@@ -110,9 +123,184 @@ async function assertTimeSeriesExist(metric, labels) {
   assert.isEmpty(resultlessQueries, `Following queries return no results: ${resultlessQueries.join(", ")}`)
 }
 
+async function retryUrl(url, redirectURL, ignoreSSl, httpStatus) {
```
Comment (crabtree) [inline tests/fast-integration/monitoring/helpers.js:126]:
```suggestion
async function retryUrl(url, redirectURL, ignoreSSL, httpStatus) {
```

File: tests/fast-integration/monitoring/helpers.js [hunk 8]
```diff
@@ -110,9 +123,184 @@ async function assertTimeSeriesExist(metric, labels) {
...
+  let retries = 0
+  while (retries < 20) {
+    let res = await queryGrafana(url, redirectURL, ignoreSSl, httpStatus)
```
Comment (crabtree) [inline tests/fast-integration/monitoring/helpers.js:129]:
```suggestion
    let res = await queryGrafana(url, redirectURL, ignoreSSL, httpStatus)
```

File: tests/fast-integration/monitoring/helpers.js [hunk 9]
```diff
@@ -110,9 +123,184 @@ async function assertTimeSeriesExist(metric, labels) {
...
+    }
+    await sleep(5*1000)
+    retries++
+  }
+  return false
+}
+
+async function assertGrafanaredirect(redirectURL) {
+  let vs = await getVirtualService("kyma-system", "monitoring-grafana")
+  let ignoreSSl = false
```
Comment (crabtree) [inline tests/fast-integration/monitoring/helpers.js:141]:
```suggestion
  let ignoreSSL = false
```

File: tests/fast-integration/monitoring/helpers.js [hunk 10]
```diff
@@ -110,9 +123,184 @@ async function assertTimeSeriesExist(metric, labels) {
...
+  if (vs.includes("local.kyma.dev")) {
+    ignoreSSl = true
```
Comment (crabtree) [inline tests/fast-integration/monitoring/helpers.js:143]:
```suggestion
    ignoreSSL = true
```

File: tests/fast-integration/monitoring/helpers.js [hunk 11]
```diff
@@ -110,9 +123,184 @@ async function assertTimeSeriesExist(metric, labels) {
...
+  }
+  let url = "https://"+vs
+  if (redirectURL.includes("https://dex.")) {
+    console.log("Checking redirect for dex")
+    return await retryUrl(url, redirectURL, ignoreSSl, 200)
```
Comment (crabtree) [inline tests/fast-integration/monitoring/helpers.js:148]:
```suggestion
    return await retryUrl(url, redirectURL, ignoreSSL, 200)
```

File: tests/fast-integration/monitoring/helpers.js [hunk 12]
```diff
@@ -110,9 +123,184 @@ async function assertTimeSeriesExist(metric, labels) {
...
+  }
+
+  if (redirectURL.includes("https://kyma-project.io/docs")) {
+    console.log("Checking redirect for kyma docs")
+    return await retryUrl(url, redirectURL, ignoreSSl, 403)
```
Comment (crabtree) [inline tests/fast-integration/monitoring/helpers.js:153]:
```suggestion
    return await retryUrl(url, redirectURL, ignoreSSL, 403)
```

File: tests/fast-integration/monitoring/helpers.js [hunk 13]
```diff
@@ -110,9 +123,184 @@ async function assertTimeSeriesExist(metric, labels) {
...
+  }
+
+  if (redirectURL.includes("https://accounts.google.com/signin/oauth")) {
+    console.log("Checking redirect for google")
+    return await retryUrl(url, redirectURL, ignoreSSl, 200)
```
Comment (crabtree) [inline tests/fast-integration/monitoring/helpers.js:158]:
```suggestion
    return await retryUrl(url, redirectURL, ignoreSSL, 200)
```

File: tests/fast-integration/monitoring/helpers.js [hunk 14]
```diff
@@ -110,9 +123,184 @@ async function assertTimeSeriesExist(metric, labels) {
...
+  }
+
+  if (redirectURL.includes("grafana")) {
+    console.log("Checking redirect for grafana")
+    return await retryUrl(url, redirectURL, ignoreSSl, 200)
```
Comment (crabtree) [inline tests/fast-integration/monitoring/helpers.js:163]:
```suggestion
    return await retryUrl(url, redirectURL, ignoreSSL, 200)
```

**Review by crabtree — APPROVED** [review]

**Review by crabtree — APPROVED** [review]

**Pull request merged**